iib/web/errors.py

```diff
@@ -32,3 +34,45 @@ def json_error(error):
         response = jsonify({'error': msg})
         response.status_code = status_code
     return response
+
+
+def handle_broker_error(request_id, request_model):
+    """
+    Handle RabbitMQ error by sating request as failed and raises IIBError.
+
+    :param int request_id: Database ID of an failed request
+    :param Request request_model: type of an index image build request
+    """
+    req = db.session.query(request_model).get(request_id)
+
+    req.add_state("failed", "The scheduling of the request failed")
+    db.session.commit()
+    messaging.send_message_for_state_change(req)
+
+    current_app.logger.exception(f"The scheduling of the build request with ID {req.id} failed")
+
+    raise IIBError(f"The scheduling of the build request with ID {req.id} failed")
+
+
+def handle_broker_batch_error(requests, processed_request_ids):
+    """
+    Handle error, sets requests as failed and raises IIBError.
+
+    :param list requests: type of an index image build request
+    :param list processed_request_ids: list of processed request IDs
+    """
+    failed_ids = []
+    for req in requests:
+        if str(req.id) in processed_request_ids:
+            continue
+
+        failed_ids.append(str(req.id))
+        req.add_state('failed', 'The scheduling of request failed')
+        messaging.send_message_for_state_change(req)
+
+    db.session.commit()
+    current_app.logger.exception(
+        f'The scheduling of the build requests with IDs {", ".join(failed_ids)} failed'
```

Comment:
Could you please assign this string to a variable to avoid code duplication and doing the string substitution twice?